playbooks/app_dev/APM_guidance.adoc

```diff
@@ -0,0 +1,156 @@
+
+==== Installing New Relic on OpenShift
+
+https://blog.openshift.com/using-new-relic-to-monitor-applications-on-openshift/[https://blog.openshift.com/using-new-relic-to-monitor-applications-on-openshift/]
+
+==== Installing Sysdig on OpenShift
+
+https://sysdig.com/openshift-monitoring/[https://sysdig.com/openshift-monitoring/] +
+https://support.sysdig.com/hc/en-us/articles/211421063-Sysdig-Install-OpenShift[https://support.sysdig.com/hc/en-us/articles/211421063-Sysdig-Install-OpenShift] +
+https://blog.openshift.com/openshift-ecosystem-using-sysdig-monitor-openshift/[https://blog.openshift.com/openshift-ecosystem-using-sysdig-monitor-openshift/]
+
+==== Installing Instana on OpenShift
+
+https://blog.openshift.com/deploying-instana-apm-natively-openshift-environment/[https://blog.openshift.com/deploying-instana-apm-natively-openshift-environment/] +
+https://instana.atlassian.net/wiki/spaces/DOCS/pages/15630376/Docker+scheduled#Docker(scheduled)-OpenShift[https://instana.atlassian.net/wiki/spaces/DOCS/pages/15630376/Docker+scheduled#Docker(scheduled)-OpenShift]
+
+== Distributed Tracing
+
+Distributed tracing is about understanding how time is spent between all of the hops a request goes through in order to be completed. Distributed tracing is not new but it becomes more relevant in a microservices architecture as the average number of hops per request increases. 
+
+The current distributed tracing standard is http://opentracing.io/[OpenTracing], which has been https://www.cncf.io/blog/2016/10/11/opentracing-joins-the-cloud-native-computing-foundation/[recently accepted] by the https://www.cncf.io/[CNCF] (thus is likely to be widely embraced).
+
+The reference implementation of OpenTracing is https://github.com/jaegertracing/jaeger[Jaeger] (also part of CNCF).
+
+The general architecture of Jaeger is the following:
+
+image::images/jaeger_architecture.png[]
+
+A jaeger client (http://jaeger.readthedocs.io/en/latest/client_libraries/[client libraries] for various languages exist) will generate trace information and sends them via a local UDP to a jaeger agent, running as a http://blog.kubernetes.io/2015/06/the-distributed-system-toolkit-patterns.html[sidecar]. The jaeger agent will perform the necessary sampling and throttling, and then send the traces to the central jaeger collector, which stores them in http://cassandra.apache.org/[Cassandra]. Once the traces are stored they can be visualized using the UI.
+
+Instructions on how to install jaeger in OpenShift can be found https://github.com/jaegertracing/jaeger-openshift[here].
+
+The UI presents result as follows:
+
+image::images/jaeger_ui.png[]
+
+If you are building a http://blog.christianposta.com/microservices/application-network-functions-with-esbs-api-management-and-now-service-mesh/[service mesh] with https://istio.io/[istio], consider that istio is naturally integrated with jaeger, as explained https://istio.io/docs/tasks/telemetry/distributed-tracing.html[here].
```

Comment:
Istio OpenShift reference links:
[1] https://blog.openshift.com/red-hat-istio-launch/
[2] https://blog.openshift.com/running-istio-service-mesh-openshift/
[3] https://blog.openshift.com/evaluate-istio-openshift/
[4] https://blog.openshift.com/istio-traffic-management-diving-deeper/
[5] https://blog.openshift.com/openshift-commons-briefing-103-microservices-istio-openshift-chak-jonagam-red-hat/

Reply:
what should I do with these links? This article is not about istio.